src/resources/views/crud/fields/video.blade.php

```diff
@@ -169,42 +170,57 @@
                 crossDomain: true,
                 success: function (data) {
                     if (typeof(data.items[0]) != "undefined") {
-                                    var v = data.items[0].snippet;
-
-                                    video.id = videoId;
-                                    video.title = v.title;
-                                    video.image = v.thumbnails.maxres ? v.thumbnails.maxres.url : v.thumbnails.default.url;
-                                    video.url = 'https://www.youtube.com/watch?v=' + video.id;
+                        var s = data.items[0].snippet;
+                        var c = data.items[0].contentDetails;
+
+                        video.title = s.title;
+                        video.image = s.thumbnails.maxres ? s.thumbnails.maxres.url : s.thumbnails.default.url;
+                        video.url = `https://www.youtube.com/watch?v=${video.id}`;
+
+                        var regExp = /PT(\d+)(?:H|M|S)(\d+)*(?:M|S)*(\d+)*(?:S)*/;
+                        var match = c.duration.match(regExp);
+                        if (match[3] !== undefined) {
+                            video.duration = parseInt(match[3]) + (parseInt(match[2]) * 60) + (parseInt(match[1]) * 3600);
+                        } else if (match[2] !== undefined) {
+                            video.duration = parseInt(match[2]) + (parseInt(match[1]) * 60);
+                        } else if (match[1] !== undefined) {
+                            video.duration = parseInt(match[1]);
+                        }
```

Comment:
From what I understood the format is `1H20M40S` or `1M40S`, so something like this would work;
```suggestion
                        video.duration = c.duration
                            .match(/\d+/g)
                            .reverse()
                            .map((e, i) => e * 60 ** i)
                            .reduce((a, b) => a + b);
```

Reply:
Ah sweet that is much cleaner, I like it! I'm gonna run some tests with it to see if it indeed covers all possible formats. Will report back to you!